scheduler/test/cook/test/cors.clj

```diff
@@ -0,0 +1,92 @@
+(ns cook.test.cors
+  (:use clojure.test)
+  (:require [clojure.string :as str]
+            [cook.cors :as cors]))
+
+
+(deftest test-is-preflight?
+  (is (cors/preflight? {:request-method :options}))
+  (is (not (cors/preflight? {:request-method :get})))
+  (is (not (cors/preflight? {:request-method :put}))))
+
+(deftest test-same-origin?
+  (is (cors/same-origin? {:headers {"origin" "http://example.com"
+                                    "host" "example.com"}
+                          :scheme :http}))
+  (is (not (cors/same-origin? {:headers {"host" "example.com"}
+                               :scheme :http})))
+  (is (not (cors/same-origin? {:headers {"origin" "http://bad.example.com"
+                                         "host" "example.com"}
+                               :scheme :http})))
+  (is (not (cors/same-origin? {:headers {"origin" "http://example.com"
+                                         "host" "example.com"}
+                               :scheme :https}))))
+
+(deftest test-request-allowed?
+  (let [allowed-orgins [#"^https?://example.com$"
+                        #"^https://secure.example.com"]
+        origin-allowed? (fn [origin] (cors/request-allowed? {:headers {"origin" origin}}
+                                                           allowed-orgins))]
+    (is (origin-allowed? "http://example.com"))
+    (is (origin-allowed? "https://example.com"))
+    (is (origin-allowed? "https://secure.example.com"))
+    (is (not (origin-allowed? "http://secure.example.com")))
+    (is (cors/request-allowed? {:headers {"origin" "http://same.example.com"
+                                          "host" "same.example.com"}
+                                :scheme :http}
+                               []))))
```

Comment:
Can we include an example where the origin and host headers are different?

Reply:
I added a host to the `origin-allowed?` headers so that cover should cover it.